src/Raven.Server/Documents/RetireAttachmentsSender.cs

```diff
@@ -308,19 +229,62 @@ internal async Task<int> RetireAttachments(int batchSize, long maxItemsToProcess
             return totalCount;
         }
 
-        private static string BuildFailureDetails(List<UploadAttachmentException> exceptions)
+        private async Task CreateUploadTaskAsync(AttachmentUploader uploader, ConcurrentQueue<AttachmentRetirementInfo> concurrentItems, AttachmentRetirementInfo attachment)
         {
-            if (exceptions.Count == 0)
-                return string.Empty;
+            using (_database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            using (_database.DocumentsStorage.AttachmentsStorage.RetiredAttachmentsStorage.ExtractHashSliceFromAttachmentId(context, attachment.Key, out Slice hashSlice))
+            using (context.OpenReadTransaction())
+            {
+                var hash = hashSlice.ToString();
+                var attachmentStream = GetAttachmentStreamAndStreamLength(context, hashSlice, await uploader.GetObjectSizeAsync(string.Empty, hash), out var attachmentLength);
+                
+                if (attachmentStream == null)
+                {
+                    // attachment was deleted or already exist in cloud, need to remote it from retired tree
```

Comment:
I don't understand the comment

Reply:
done, this means the attachments exists on remote destination, or doesn't exist anymore